manager/integration/framework/node_exec/node_exec.py

```diff
@@ -0,0 +1,197 @@
+            cmd
+        ]
+        res = stream(
+            self.core_api.connect_get_namespaced_pod_exec,
+            pod.metadata.name,
+            self.namespace,
+            command=exec_command,
+            stderr=True,
+            stdin=False,
+            stdout=True,
+            tty=False
+        )
+        return res
+
+    def launch_pod(self, node_name):
+        if node_name in self.node_exec_pod:
+            return self.node_exec_pod[node_name]
+        else:
+            pod_manifest = {
+                'apiVersion': 'v1',
+                'kind': 'Pod',
+                'metadata': {
+                    'name': node_name
+                },
+                'spec': {
+                    'affinity': {
+                        'nodeAffinity': {
+                            'requiredDuringSchedulingIgnoredDuringExecution': {
+                                'nodeSelectorTerms': [{
+                                    'matchExpressions': [{
+                                        'key': 'kubernetes.io/hostname',
+                                        'operator': 'In',
+                                        'values': [
+                                            node_name
+                                        ]
+                                    }]
+                                }]
+                            }
+                        }
+                    },
+                    'containers': [{
+                        'image': 'busybox:1.34.0',
+                        'imagePullPolicy': 'IfNotPresent',
+                        'securityContext': {
+                            'privileged': True
+                        },
+                        'name': 'node-exec',
+                        "args": [
+                            "tail", "-f", "/dev/null"
+                        ],
+                        "volumeMounts": [{
+                            'name': 'rootfs',
+                            'mountPath': '/rootfs',
+                            'readOnly': True
+                        }],
+                    }],
+                    'volumes': [{
+                        'name': 'rootfs',
+                        'hostPath': {
+                            'path': '/'
+                        }
+                    }]
+                }
+            }
+            pod = self.core_api.create_namespaced_pod(
+                body=pod_manifest,
+                namespace=self.namespace
+            )
+            for i in range(DEFAULT_POD_TIMEOUT):
+                pod = self.core_api.read_namespaced_pod(
+                        name=node_name,
+                        namespace=self.namespace
+                      )
+                if pod is not None and pod.status.phase == 'Running':
+                    break
+                time.sleep(DEFAULT_POD_INTERVAL)
+            self.node_exec_pod[node_name] = pod
+            return pod
+    
+    def power_off_node(self, node_name):
```

Comment:
Can these operations (power_off_node, power_on_node, restart_kubelet, etc.) be moved to another class? Since this class is for running a pod on a specific node and not related to SSH or other instance operations.

Reply:
Understood, I plan to move these operations under `node` directory, then it will be like this:
<img width="400" alt="截圖 2023-05-03 上午8 56 55" src="https://user-images.githubusercontent.com/17548901/235815223-5c87673f-f1a9-4d15-be80-9489a01ee73b.png">